### ISSUE TITLE
Some clarity on deployment behind Nginx as proxy server

### ISSUE THREAD

**Issue by seanmavley** [strapi/strapi#2617]
I'm a few hours old with Strapi. Setup locally in seconds. Trying to deploy with Nginx sitting in front as proxy. Totally different experience.

My setup:
18.04 LTS | Node 11 | MongDB 4.0.5 | Nginx 1.14 | Strapi 3.0.0-alpha.18

What I want to achieve (or so I thought):
 - Run Nginx on port 80/443
 - Proxy requests hitting api.example.com to Strapi (all of strapi)
   - so api.example.com/ should take me to the index. Then /admin should take me to the admin
 - That's it.

What I've done:
So I saw this article, https://blog.strapi.io/how-to-deploy-a-strapi-application/ (pretty vague enough for a real-world deployment setup)

Guessing (plentiful) from the above article, I ended up with this:

```
upstream api_example {
    server 127.0.0.1:1337;
}

server {
    listen 443 ssl http2;

    server_name api.example.com;

    include /etc/nginx/ssl/ssl-params.conf;
    include /etc/nginx/ssl/example.com.ssl;

    error_log /var/log/nginx/api.example.com_error.log warn;
    access_log /var/log/nginx/api.example.com_access.log;

    root /home/khophi/Strapi; # this is root folder, after running `strapi new Strapi && cd Strapi`

    location / {
        proxy_pass http://api_example/;
	proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_http_version 1.1;
        proxy_set_header X-NginX-Proxy true;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_cache_bypass $http_upgrade;
	proxy_redirect off;
	proxy_set_header X-Forwarded-Proto $scheme;
    }
}

server {
    listen 80;
    server_name api.example.com;
    return 301 https://$server_name$request_uri?;
}
```
Obviously, the above fails.

Going to / works fine, and shows me the static `index.html`. Going /admin fails. Blank white screen with 2 failed static loads in network tab in Dev Tools, namely, `https://api.example.com/vendor.dll.js` and `https://api.example.com/main.js`

I'm not doing something right.

I've seen this too: https://strapi.io/documentation/3.x.x/advanced/customize-admin.html#deployment, except none of that mention anything how to go about with Strapi behind a Proxy, like Nginx.

To be honest, I'm getting more confused the more I search for answers, each one saying something else.

In summary
 - Nginx as proxy server
 - API at api.example.com
 - admin at api.example.com/admin

Will really appreciate some definite real-world deployment examples.

**Comment by derrickmehaffy** [comment]
@seanmavley https://medium.com/@derrickmehaffy/using-strapi-in-production-with-https-717af82d1445

**Comment by derrickmehaffy** [comment]
Specifically your proxy block and you have http2 enabled (which nginx doesn't support via the proxy)

Also as a personal preference I recommend dumping your upstreams into it's own file like I do in that guide.

**Comment by cpres** [comment]
@seanmavley  would be interested to hear exactly what you changed, Derrick's article didn't help  me but looks like you got past your problem

**Comment by hnq90** [comment]
I also got same issue with @seanmavley.
I have Apache reverse proxy which proxy request to local Strapi server by Docker.

When I go to: https://example.com/, the index was displayed successfully but https://example.com/admin got blank screen.

**Comment by juliocesarpg92** [comment]
Hello, i already ran `npm run setup`, and followed the guide from @derrickmehaffy and when i try to access to /admin i get 404 error, i use strapi 3.0.0-alpha.26.2. 
Regards

**Comment by derrickmehaffy** [comment]
@juliocesarpg92 alpha is no longer supported

**Comment by euuujn** [comment]
@seanmavley did you find solution? I stuck with same problem 🤯

**Comment by jeethjoseph** [comment]
I can access my admin page **ip:port/admin** but when I try to proxy pass it through nginx **example.com/admin** get to the admin page but the loader keeps circling without end. I can see that I'm on the right page because the page title shows up in the browser tab.
Here's my config block.
` location /admin {
          proxy_set_header   X-Forwarded-For $remote_addr;
          proxy_set_header   Host $http_host;
          proxy_pass http://127.0.0.1:**port**/admin;
      }
`

**Comment by derrickmehaffy** [comment]
@jeethjoseph a PR is being worked on for sub-folder based proxying, for the time being you will need to use a sub-domain until this is merged: https://github.com/strapi/strapi/pull/5724
